src/Block.cpp

```diff
@@ -32,19 +33,79 @@ std::string Block::ToString() const {
  */
 bool Block::isFreeBlock() const {
 	FreeBlock* fb = (FreeBlock*) this;
-	return (void*) fb->typeDescriptor == fb->dataPosition();
+	FreeBlock fbA = *fb;
+	TypeDescriptor* td = fb->getTypeDescriptor();
+	void* dataPart = fb->getDataPart();
+	return (void*) fb->getTypeDescriptor() == fb->dataPosition();
 }
 void* Block::getDataPart() const {
 	// "this + sizeof(Block)" is the effective getDataPart for the block
-	return (void*) (this + headerSize());
+	return (void*) ((char*) this + headerSize());
```

Comment:
This cast is needed, otherwise it is basically `this + headerSize() * sizeof(Block)`.
Meaning, the pointer would be too far off. With the `(char*)` cast it is `headerSize() * sizeof(char*))` which results in just `headerSize()`